library/kani_macros/src/lib.rs

```diff
@@ -49,3 +50,29 @@ pub fn proof(_attr: TokenStream, item: TokenStream) -> TokenStream {
     //     $item
     // )
 }
+
+#[cfg(not(kani))]
+#[proc_macro_attribute]
+pub fn unwind(_attr: TokenStream, _item: TokenStream) -> TokenStream {
+    // Not-RMC, Not-Test means this code shouldn't exist, return nothing.
+    TokenStream::new()
+}
+
+#[cfg(kani)]
+#[proc_macro_attribute]
+pub fn unwind(attr: TokenStream, item: TokenStream) -> TokenStream {
+    let mut result = TokenStream::new();
+
+    // Translate #[kani::unwind(arg)] to #[kanitool::unwind_arg_] for easier handling
+    let insert_string = "#[kanitool::unwind_".to_owned() + &attr.clone().to_string() + "]";
+
+    // Add the string that looks like - #[kanitool::unwind_value_]
+    result.extend(insert_string.parse::<TokenStream>().unwrap());
+    // No mangle seems to be necessary as removing it prevents all the attributes in a lib from being read
```

Comment:
I don't understand this comment. Unwind should be on something that already has `#[proof]` surely?

Reply:
This was a design choice made to separate unwind as an attribute from proof i.e, the user should be able to call unwind on harnesses without the proof attribute. This might be a disagreement in design that should be resolved before we add other attributes as well.